### PR TITLE
boards: nucleo_l552ze_q_ns: Use dts flash partitioning.

### PR DESCRIPTION
**Pull request by erwango**
Provide device tree flash partition in order to configure
and flash non secure binary.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

### CODE REVIEW

File: boards/arm/nucleo_l552ze_q/nucleo_l552ze_q-common.dtsi [hunk 1]
```diff
@@ -32,3 +32,36 @@
...
+		#address-cells = <1>;
+		#size-cells = <1>;
+
+		boot_partition: partition@0 {
+			label = "mcuboot";
+			reg = <0x00000000 0x00020000>;
+			read-only;
+		};
+		/* Secure image primary slot */
+		slot0_partition: partition@00011000 {
```
Comment (ABOSTM) [inline boards/arm/nucleo_l552ze_q/nucleo_l552ze_q-common.dtsi:49]: It looks like there is an overlap with boot_partition which end at 0x00020000

**Review by ABOSTM — CHANGES_REQUESTED** [review]

**Comment by erwango** [comment]
@ioannisg moved partition as an overlay as it depends on TFM configuration.
Also I modified the image size (rounded). Reason is that it doesn't depends only on the binary start address, so having 400 bytes more made things unclear.

**Comment by ioannisg** [comment]
> @ioannisg moved partition as an overlay as it depends on TFM configuration.
> Also I modified the image size (rounded). Reason is that it doesn't depends only on the binary start address, so having 400 bytes more made things unclear.

This is not ideal, either, @erwango because the overlay should not be sample-specific, i.e. it should be the same for any build on this Nucleo board, with TFM enabled. I believe we need a more generic solution. Perhaps an overlay for this board that is taken into when we build with TFM....

**Comment by erwango** [comment]
> This is not ideal, either, @erwango because the overlay should not be sample-specific, i.e. it should be the same for any build on this Nucleo board, with TFM enabled. 

Overlay are essentially sample specific.. I agree we can argue when overlay are used to change the flash partitioning. But even then, we can think of applications having specific needs (large storage, no storage at all, ...).

Anyway, I think that with TFM configuration scheme, it is actually sample specific and we have no choice. 
`tfm_ipc` requires enabling TFM option `IPC_TEST`. This embeds a lot of additional code and bloats secure image (246K in TFMLevel2 !), which is not compatible with having primary and secondary images on limited (512K) flash of the board.
Not using this config (`ConfigCoreIPCTfmLevel2`), we're able to fit 2 sets of primary and secondary images.

One thing that could be done is to provide a "generic" flash partitioning (primary S/NS, secondary S/NS) for this board ... which would be overwritten by proposed overlay.

File: boards/arm/nucleo_l552ze_q/nucleo_l552ze_q-common.dtsi (hunk 1 above)
Comment (erwango) [reply boards/arm/nucleo_l552ze_q/nucleo_l552ze_q-common.dtsi]: Good catch, fixed

File: boards/arm/nucleo_l552ze_q/Kconfig.defconfig [hunk 2]
```diff
@@ -8,4 +8,11 @@ if BOARD_NUCLEO_L552ZE_Q
 config BOARD
 	default "nucleo_l552ze_q"
 
+if TRUSTED_EXECUTION_NONSECURE
+
+config USE_DT_CODE_PARTITION
+	default y
+	# Get flash configuration for NS image from dts flash partition
+endif
```
Comment (ioannisg) [inline boards/arm/nucleo_l552ze_q/Kconfig.defconfig:16]: add an inline comment # TRUSTED_EXECUTION_NONSECURE

**Review by ioannisg — APPROVED** [review]
Looks OK to me.
Left some minor style comments.

File: boards/arm/nucleo_l552ze_q/Kconfig.defconfig [hunk 3]
```diff
@@ -8,4 +8,11 @@ if BOARD_NUCLEO_L552ZE_Q
 config BOARD
 	default "nucleo_l552ze_q"
 
+if TRUSTED_EXECUTION_NONSECURE
+
+config USE_DT_CODE_PARTITION
+	default y
+	# Get flash configuration for NS image from dts flash partition
```
[the remark this answers is not in the document]
Comment (erwango) [reply boards/arm/nucleo_l552ze_q/Kconfig.defconfig]: Indeed, fixed.

File: boards/arm/nucleo_l552ze_q/Kconfig.defconfig (hunk 2 above)
Comment (erwango) [reply boards/arm/nucleo_l552ze_q/Kconfig.defconfig]: Fixed

**Review by ABOSTM — APPROVED** [review]
LGTM

**Pull request merged**